src/uct/ib/dc/dc_mlx5_ep.h

```diff
@@ -523,10 +542,7 @@ uct_dc_mlx5_iface_dci_get(uct_dc_mlx5_iface_t *iface, uct_dc_mlx5_ep_t *ep)
         return UCS_OK;
     }
 
-    /* Do not alloc dci if no TX desc resources,
-     * otherwise this dci may never be released. */
-    if (uct_dc_mlx5_iface_dci_can_alloc(iface, pool_index) &&
-        uct_dc_mlx5_iface_has_tx_resources(iface)) {
```

Comment:
I see, but before this PR didn't we already check RDMA_READ credits before getting a DCI?

Reply:
yes, but I meant the check for `ucs_mpool_is_empty(&iface->super.super.tx.mp)` inside `uct_dc_mlx5_iface_has_tx_resources`
Currently we do not check it in send flow as an optimization

Reply:
you mean that instead of just removing uct_dc_mlx5_iface_has_tx_resources, as I did in this PR, we should make sure to check  `ucs_mpool_is_empty(&iface->super.super.tx.mp)` ?